backfill_mismatched_deposits.sql

```diff
@@ -0,0 +1,64 @@
+-- soft-delete events for overcounted deposits
+UPDATE events SET deleted_at = current_timestamp
+WHERE
+  deposit_id IN (
+  SELECT 
+    deposits.id
+  FROM
+    deposits
+  LEFT JOIN 
+    blocks
+  ON
+    deposits.block_hash = blocks.hash
+  WHERE
+    deposits.main = true AND (blocks.main IS NULL OR blocks.main = false)
+);
+
+-- upsert events for undercounted mismatched deposits
+INSERT INTO events (type, occurred_at, points, user_id, deposit_id)
```

Comment:
Just want to make sure that this updates instead of creating new events. Because we have the [event.id as auto-increment](https://github.com/iron-fish/ironfish-api/blob/61b1cde0b82e412ab691ddc90c5e9e2ff2c27cac/prisma/schema.prisma#L12). Was wondering if `UPDATE` maybe isn't better for this query since I can't think of scenarios where the event would not already be in the table

Reply:
Ah good call, there needs to be another clause in this query with `ON CONFLICT` to address updates. There's a uniqueness constraint on `deposit_id`, so this will only insert events for deposits where there isn't one already.